_tasks/util/gh-file-importer.mjs

```diff
@@ -0,0 +1,141 @@
+/**
+ * @file Imports abritrary files from remote GitHub repositories.
+ * @author Derek Lewis <DerekNonGeneric@inf.is>
+ */
+
+// -----------------------------------------------------------------------------
+// Requirements
+// -----------------------------------------------------------------------------
+
+import { createRequire } from 'module';
+import { inspect } from 'util';
+import {
+  basename as pathBasename,
+  join as pathJoin,
+} from 'path';
+import { writeFileSync } from 'fs';
+import axios from 'axios';
+import axiosDebugLog from 'axios-debug-log';
+
+const require = createRequire(import.meta.url);
+const log = require('fancylog');
+const { Octokit } = require('@octokit/rest');
+
+// -----------------------------------------------------------------------------
+// Public API
+// -----------------------------------------------------------------------------
+
+export class GhFileImporter {
+  /**
+   * Creates an instance of GhFileImporter.
+   * @param {string} destDir The destination directory.
+   * @memberof GhFileImporter
+   */
+  constructor(destDir) {
+    if (!destDir) {
+      throw new Error('A destination directory must be supplied.');
+    }
+    this._log = log;
+    // Add an API key if the GitHub API throttling gets you.
+    this._octokit = new Octokit({ auth: `${process.env.GITHUB_SECRET}` });
+    this._destDir = destDir;
+  }
+
+  /**
+   * Retrieves a path's metadata from the GitHub API.
+   * @param {string} owner The username associated with the repository.
+   * @param {string} repo The repository's name.
+   * @param {!(string | undefined)} path Path to the file or folder.
+   * @param {!(string | undefined)} ref The name of the commit/branch/tag.
+   *     Default: the repository's default branch (usually master).
+   * @returns {Promise<Object>} An object containing all information.
+   * @memberof GhFileImporter
+   */
+  _fetchPathMetadata(owner, repo, path = undefined, ref = undefined) {
+    if (path === undefined) {
+      return this._octokit.repos.getContent({
+        owner,
+        repo,
+      });
+    } else if (ref === undefined) {
+      return this._octokit.repos.getContent({
+        owner,
+        repo,
+        path,
+      });
+    }
+  }
+
+  /**
+   * Validates a request to retrieves a path's metadata prior to doing so.
+   * @public
+   * @param {string} owner The username associated with the repository.
+   * @param {string} repo The repository's name.
+   * @param {!(string | undefined)} path Path to the file or folder.
+   * @param {!(string | undefined)} ref The name of the commit/branch/tag.
+   *     Default: the repository's default branch (usually master).
+   * @returns {Promise<Object>} An object containing all information.
+   * @memberof GhFileImporter
+   */
+  async fetchPathMetadata(owner, repo, path = undefined, ref = undefined) {
+    if (!owner) {
+      return Promise.reject(new Error('Cannot download from undefined owner.'));
```

Comment:
alternatively:
```suggestion
      throw new Error('Cannot download from undefined owner.');
```

Reply:
This seems like something that should be made into a lint rule. A cursory look turned up the [`prefer-promise-reject-errors`](https://eslint.org/docs/rules/prefer-promise-reject-errors) ESLint rule, which looks like a nice-to-have, but also would be irrelevant if we want to prefer throwing instead.